taichi/program/context.h

```diff
@@ -32,6 +34,8 @@ struct RuntimeContext {
   uint64 array_runtime_sizes[taichi_max_num_args_total]{0};
   // `is_device_allocations` is true iff i-th arg is a `DeviceAllocation*`.
   bool is_device_allocations[taichi_max_num_args_total]{false};
```

Comment:
can we remove `is_device_allocation` now? (Maybe we need to add something like `DevAllocType::kInvalid`.